### PR TITLE
MdePkg/IpmiNetFnGroupExtension.h: Enforce structure alignment

### PR DESCRIPTION
**Pull request by nhivp**
# Description

The natural aligmenent seems to be failed on some cases. So, this patch intends to add the pack(1) to ensure the structure aligned with a one-byte boundary.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

### CODE REVIEW

File: MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #include <Pi/PiStatusCode.h>
 
+#pragma pack(1)
```
Comment (mdkinney) [inline MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h:14]: I do not see a specification reference in the file header.  Can one be added that contains a clear description of the structure layout to make it clear by byte packing is required?

Comment (nhivp) [reply MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h:14]: Hi @mdkinney, the ARM Server Base System Architecture (SBSA) specification, version 2.0, is referenced on line 27. Section F clearly defines the offset for each field within the structure. As the structure does not include padding, alignment should be ensured with 1-byte packing.

Should I move the `#pragma pack(1)` after line 27 or include the specification reference in the file header for clarity?

Comment (mdkinney) [reply MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h:14]: Yes.  File header reference is required.  The #pragma pack(1) should only be around structure declarations that are required to be byte package by the specification.

Comment (nhivp) [reply MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h]: Thanks @mdkinney for the clarification. I've updated it.

**Review by mdkinney — APPROVED** [review]

**Pull request merged**